### PR TITLE
Implement Connecting view at top of list

### LINKED ISSUE #1463: [IMPROVEMENT] Connecting view hides the chat room name
The connecting view is currently hiding the chat room name as show bellow:

sc

It'd be better to don't hide that chat room name

### PR DESCRIPTION
**Pull request by shubhsherl**
@RocketChat/android

Closes #1463 

Move connecting view to top of list in chatroomFragment and chatroomsFragment.
Added color based on Connectivity status:
Connected, Authenticating -> green
Disconnected, Disconnecting -> red
Waiting, Connecting -> orange

### CODE REVIEW

**Comment by leonardoaramaki** [comment]
I'm not sure if setting different colors to states is a good idea. Too much information I think. 
Can you attach a gif showing the states transitions?

**Comment by shubhsherl** [comment]
Video can be found [here](https://youtu.be/VvY-4xzELpU).

File: app/src/main/res/layout/fragment_chat_rooms.xml [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
     <androidx.recyclerview.widget.RecyclerView
         android:id="@+id/recycler_view"
+        android:layout_below="@+id/connection_status_text"
```
Comment (philipbrito) [inline app/src/main/res/layout/fragment_chat_rooms.xml]: Should be `text_connection_status` instead.

Comment (shubhsherl) [reply app/src/main/res/layout/fragment_chat_rooms.xml]: `id/connection_status_text` was defined earlier. Should I rename it?

Comment (philipbrito) [reply app/src/main/res/layout/fragment_chat_rooms.xml]: Oh, I'm sorry! If you can, please change it then!
Thanks @shubhsherl 👍

**Comment by philipbrito** [comment]
@shubhsherl What do you think about putting that status on toolbar? A good reference is how Telegram works when you are offline.

Btw, could you change the status background color, i.e leaving it as before?

Thanks.

**Comment by philipbrito** [comment]
@shubhsherl Let me know if you are planning to put that state on toolbar and this PR will close https://github.com/RocketChat/Rocket.Chat.Android/issues/1448 too.

**Comment by shubhsherl** [comment]
@filipedelimabrito Putting the status on the toolbar doesn't look good. And for the waiting issue, we can reduce the waiting time, rather than increasing time by multiples of 3, we can put an upper bound, or don't wait at all.
For background color, I will change it to black, as it was before.

**Comment by philipbrito** [comment]
Let me see when you change the background color and we'll merge this PR.

**Comment by shubhsherl** [comment]
I was waiting for your response over the waiting time issue. Should I change it?

**Comment by shubhsherl** [comment]
@filipedelimabrito I messed up with the branch history so I am closing this PR and creating another PR.

**Pull request closed without merging**